### DISCUSSION TITLE
Warning raised when trying to extract parasitics

### DISCUSSION THREAD

**Discussion by lustefan97** [The-OpenROAD-Project/OpenROAD#4231] [Q&A]
Hello,

I have some questions about a warning that I stumble upon while trying to use the `extract_parasitics` function with the rcx_pattern.rules file.  I have an adder of variable size with two input words and one output word, the nets linked to the output word are raising some warnings while those for the input words do not.

The warning types are the following : 

[WARNING RCX-0114] Net 282 S[122] does not start from an iterm or a bterm.

But the thing is that when I  highlight this net in the gui, it seems to be connected to a pin on the top layer from the first metal layer (I have hidden the Pad layer to make the name readable) : 

<img width="896" alt="Capture d’écran 2023-11-08 à 18 22 39" src="https://github.com/The-OpenROAD-Project/OpenROAD/assets/61243950/e4cf3bf8-8747-452c-9f76-181aed905ffd">

I don't really understand what this warning is about since the net (and the others) do seems to be connected from front to end, and I didn't do anything particular to it compared to the input words nets.

I thus have three questions : 

1) What does this warning means, and what should I look for to solve it ?
2) What is the difference between "iterm" and "bterm" ?
3) I also have a DRT warning saying "Cut layer V0 has no default via defined", should a design have default vias ? How to define them and what are they used for ? Are they there to connect the standard cells source/gate/drain to M1 ?

**Comment by maliberty** [comment]
1) Its hard to tell whats going on with the routing in the image.  Can you provide a test case?
2) an iterm is the pin of a cell whereas a bterm is a top level pin
3) I wouldn't worry about it.  Which PDK is this?

**Reply by lustefan97** [reply]
Sure, here is a test case : 

[TestCase.zip](https://github.com/The-OpenROAD-Project/OpenROAD/files/13309404/TestCase.zip)

There is no lib file to save space for the upload but that shouldn't be a problem. Uncompress this in _OpenROAD-flow-scripts/tools/OpenROAD/test/_, use the gui and, in the tcl command box, enter : source "./scripts/constant_density/5_adder_diedim_166000_pd_0.8/adder_1024_asap7.tcl". The script should execute properly.

I am using the ASAP7 PDK for this design.
